pkg/plugins/connection/ssh.go

```diff
@@ -42,13 +48,22 @@ func (t *SSHOpen) Metadata() *gornir.TaskMetadata {
 
 // Run implements gornir.Task interface
 func (t *SSHOpen) Run(ctx context.Context, logger gornir.Logger, host *gornir.Host) (gornir.TaskInstanceResult, error) {
-	sshConfig := &ssh.ClientConfig{
-		User: host.Username,
-		Auth: []ssh.AuthMethod{
-			ssh.Password(host.Password),
-		},
-		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
-	} // #nosec
+	var sshConfig *ssh.ClientConfig
+	if t.ClientConfigFn != nil { // The client specified a config
+		config, err := t.ClientConfigFn(host, logger)
+		if err != nil {
+			return nil, err
+		}
+		sshConfig = config
+	} else { // Running with defaults
+		sshConfig = &ssh.ClientConfig{
```

Comment:
I'd take this opportunity to write a `ClientConfigFn` and do instead:

``` golang
        var clientConfigFn ClientConfigFn // Using the type I mentioned above
 	if t.ClientConfigFn != nil { // The client specified a config
              clientConfigFn = t.ClientConfigFn
        } else {
              clientConfigFn = defaultSSHClientConfig
        }
        config, err := t.ClientConfigFn(host, logger)
```

It's a bit cleaner and `defaultSSHClientConfig` would serve as an example to anyone looking at the code